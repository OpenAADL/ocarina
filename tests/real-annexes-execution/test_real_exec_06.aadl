package Test5_REAL
public

-----------------
-- Subprograms --
-----------------

subprogram Hello_Spg_1
properties
  source_language => Ada95;
  source_name     => "Hello.Hello_Spg_1";
end Hello_Spg_1;

subprogram Hello_Spg_2
properties
  source_language => Ada95;
  source_name     => "Hello.Hello_Spg_2";
end Hello_Spg_2;

-------------
-- Threads --
-------------

thread Task
end Task;

thread implementation Task.impl_1
calls
   MyCalls : {
      P_Spg : subprogram Hello_Spg_1;
   };
properties
   Dispatch_Protocol                  => periodic;
   Period                             => 1000ms;
   Compute_Execution_time             => 0 ms .. 3 ms;
   Deadline                           => 1000 ms;
   Source_Stack_Size                  => 13952 Bytes;
end Task.impl_1;

thread implementation Task.impl_2
calls
   MyCall : {
     P_Spg : subprogram Hello_Spg_2;
   };
properties
   Dispatch_Protocol                  => periodic;
   Period                             => 500ms;
   Compute_Execution_time             => 0 ms .. 3 ms;
   Deadline                           => 500 ms;
   Source_Stack_Size                  => 13952 Bytes;
end Task.impl_2;

---------------
-- Processor --
---------------

processor cpurm
end cpurm;

processor implementation cpurm.impl
properties 
   Scheduling_Protocol => (Posix_1003_Highest_Priority_First_Protocol);
end cpurm.impl;

---------------
-- Processes --
---------------

process node_a
end node_a;

process implementation node_a.impl
subcomponents
   Task1 : thread Task.impl_1;
   Task2 : thread Task.impl_2;

annex real_specification {**

--  Tests references properties extraction

theorem test_reference_property

  foreach cpu in processor_set do

  var cos_x  := cos (1.0);
  var sin_x  := sin (1.0);
  var tan_x  := tan (1.0);
  var cosh_x := cosh (1.0);
  var sinh_x := sinh (1.0);
  var tanh_x := tanh (1.0);

  var ln_x := ln (2.0);
  var exp_x := exp (1.0);
  var sqrt_x := sqrt (2.0);

  check (1 = 1);
end test_reference_property;

**};

end node_a.impl;

------------
-- System --
------------

system rma 
end rma;

system implementation rma.ERC32
subcomponents
   node_a : process node_a.impl;
   cpu_rm : processor cpurm.impl;
properties
   Actual_Processor_Binding => (reference (cpu_rm)) applies to node_a;
end rma.ERC32;

end Test5_REAL; 
