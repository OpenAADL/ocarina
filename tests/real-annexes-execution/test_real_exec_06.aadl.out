test_real_exec_06.aadl:10:03: Warning: source_language is not a list while the corresponding property name at programming_properties.aadl:61:02 is a list.
test_real_exec_06.aadl:10:03: Warning: The value of source_language has been converted into a list.
test_real_exec_06.aadl:16:03: Warning: source_language is not a list while the corresponding property name at programming_properties.aadl:61:02 is a list.
test_real_exec_06.aadl:16:03: Warning: The value of source_language has been converted into a list.
test_reference_property execution
-------------------------------------
Evaluating theorem test_reference_property

 * Iterate for variable: rma.erc32_cpu_rm
     -> value for cos_x is 0.540302305868139717
     -> value for sin_x is 0.841470984807896507
     -> value for tan_x is 1.55740772465490223
     -> value for cosh_x is 1.54308063481524378
     -> value for sinh_x is 1.17520119364380146
     -> value for tanh_x is 0.761594155955764888
     -> value for ln_x is 0.693147180559945309
     -> value for exp_x is 2.71828182845904524
     -> value for sqrt_x is 1.41421356237309505
 => Result: TRUE

theorem test_reference_property is: TRUE


